### DISCUSSION TITLE
tree node

### DISCUSSION THREAD

**Discussion by qingtingzhiyu39** [eclipse-theia/theia#14156] [General]
hi,
   I hava a question.is that I want to record the trees that have already been opened.
   I have a tree,The default setting tree's expanded is "true",then at this moment, I opened a tree node to display its own image. Then trigger a event,I have an operation to refresh the tree again,I want to keep open tree nodes while refreshing the tree. How should I do this。
   Now i refresh tree,All tree nodes have been reset ？？？

**Comment by msujew** [comment]
Hey @qingtingzhiyu39,

I believe this is a similar issue to https://github.com/eclipse-theia/theia/issues/13946. It's just currently not implemented in the tree widget to hold the expanded state of the tree node when the tree is refreshed.
